### PR TITLE
Updates Button component `loading` API

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Review by spark33 — CHANGES_REQUESTED** [review]
- The `Loading` story should be added to `meta.parameters.generate.storyNames` in `Button.story.tsx`
-

File: packages/button/src/ButtonContent/ButtonContent.tsx [hunk 1]
```diff
@@ -50,6 +50,17 @@ export const ButtonContent = (props: ButtonContentProps) => {
...
 
+  const spinner =
+    loadingIndicator &&
+    React.cloneElement(loadingIndicator, {
+      className: cx({
+        [centeredSpinnerStyles]: !loadingText,
+      }),
+      sizeOverride: buttonSpinnerSize[size],
+      colorOverride: spinnerColor[theme],
+      ['data-testid']: 'lg-button-spinner',
```
Comment (spark33) [inline packages/button/src/ButtonContent/ButtonContent.tsx:61]: Should spread other props passed to Spinner

File: packages/button/src/ButtonContent/ButtonContent.tsx [hunk 2]
```diff
@@ -50,6 +50,17 @@ export const ButtonContent = (props: ButtonContentProps) => {
     return unregisterRipple;
   }, [rippleRef, variant, darkMode, disabled, theme]);
 
+  const spinner =
+    loadingIndicator &&
+    React.cloneElement(loadingIndicator, {
+      className: cx({
+        [centeredSpinnerStyles]: !loadingText,
```
Comment (spark33) [inline packages/button/src/ButtonContent/ButtonContent.tsx:57]: Should also allow the user to pass their own className to Spinner

File: packages/button/src/ButtonContent/ButtonContent.tsx (hunk 1 above)
Comment (bruugey) [reply packages/button/src/ButtonContent/ButtonContent.tsx:65]: updated

File: packages/button/src/ButtonContent/ButtonContent.tsx (hunk 2 above)
Comment (bruugey) [reply packages/button/src/ButtonContent/ButtonContent.tsx]: updated

**Review by spark33 — APPROVED** [review]

**Pull request merged**